eng/build-utils.ps1

```diff
@@ -22,19 +22,21 @@ function GetProjectOutputBinary([string]$fileName, [string]$projectName = "", [s
   return Join-Path $ArtifactsDir "bin\$projectName\$configuration\$tfm\$ridDir$publishDir$fileName"
 }
 
-function GetPublishData() {
+function GetPublishData([string]$branchName = "main") {
```

Comment:
this now pulls publish data from the branch we're on.  this does mean that release branch configurations will have to be updated on the release branch, but that doesn't seem unreasonable to me.

this makes it 10x easier to test changes that require test branches.

Reply:
Would this also allow us to run the official build pipeline on any branch? In other words, will we be able to deprecate the "PR validation" pipeline and have one pipeline for both (like we have in Razor) after this PR?

Reply:
I'm actually going to revert this change.  there is more that is needed to actually pull the publishdata from the branch, and I'd rather do it all together (insertion still pulls from main).